src/main/java/net/atos/zac/aanvraag/ProductaanvraagService.java

```diff
@@ -365,7 +394,8 @@ private void pairProductaanvraagInfoWithZaak(
                 // there can be at most only one initiator for a particular zaak so even if there are multiple (theorically possible)
                 // we are only interested in the first one
                 .findFirst()
-                .ifPresent(betrokkene -> addInitiator(betrokkene.getInpBsn(), zaak.getUrl(), zaak.getZaaktype()));
+                .flatMap(betrokkene -> getRol(betrokkene, zaak.getUrl(), zaak.getZaaktype(), RolType.OmschrijvingGeneriekEnum.INITIATOR))
```

Comment:
I'm afraid I still cannot map this code change to what I read in the story (which is too minimal I think). For example: why now have a generic `getRol` method which is here always called with the roltype `initiator`. Then you may as well call that method `getInitiatorRole` right?

Reply:
But my first question is: what is the requirement from the story? Is it about adding initiators only or also about other role types?

Reply:
I agree that it could have been written down more clearly, but we only need to add initiators. But theoretically, other role types can be included in the. If we need to do that in the future, we can re-use this function.

Reply:
Clear. However in general I think that is not a good coding practise. You should never code for the future but only for now is my idea. :-) IntelliJ will also give you a warning with code like this? It then usually says: this method is always called with the same argument so you should refactor it.

Reply:
Personally, I don't mind it, because it doesn't make the code less clear to me. It indeed results in a warning, but I don't think having a method always called with the same argument or not is a good heuristic for code quality. Having said that, if that's in line with the coding standards, I'll change it.

Reply:
Done